File: dotnet/CardsProject/Domain/Cards/CardRuling.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace CardsProject.Domain.Cards;

public class CardRuling
{
    public int Id { get; set; }

    public string RulingText { get; set; } = "";
    public DateOnly? PublishedAt { get; set; } = null;
    public string Source { get; set; } = "";

    public int? CardId { get; set; }
    [ForeignKey(nameof(CardId))]
    public Card? Card { get; set; }

    // Business operations

    public bool IsCurrent()
    {
        // TODO: implement is_current
        return default;
    }

    public bool SupersedesPrevious()
    {
        // TODO: implement supersedes_previous
        return default;
    }
}

File: phoenix/lib/cards_project/cards/card_ruling.ex
defmodule CardsProject.Cards.CardRuling do
  use Ecto.Schema
  import Ecto.Changeset

  schema "card_rulings" do
    field :ruling_text, :string
    field :published_at, :date
    field :source, :string
    belongs_to :card, CardsProject.Cards.Card

    timestamps()
  end

  @doc false
  def changeset(record, attrs) do
    record
    |> cast(attrs, [:ruling_text, :published_at, :source, :card_id])
    |> validate_required([:ruling_text, :published_at, :source])
  end

  # ── Business operations ────────────────────────────────────────────

  def is_current(_record) do
    # TODO: implement CardRuling.is_current
    {:error, :not_implemented}
  end

  def supersedes_previous(_record) do
    # TODO: implement CardRuling.supersedes_previous
    {:error, :not_implemented}
  end
end
